src/use-css.hook.js

```diff
@@ -17,28 +17,29 @@ export function useCss(css, overrideNamespace) {
 
   function useStyleElement() {
     useLayoutEffect(() => {
-      const newStyleElement = getStyleElement(styleElement, isPostCss, css, namespace)
+      const newStyleElement = getStyleElement(isPostCss, css, namespace, false)
       setStyleElement(newStyleElement)
 
       return () => {
         if (--styleElement.kremlings === 0) {
+          const rawCss = isPostCss ? css.styles : css
           document.head.removeChild(styleElement)
-          delete styleTags[css]
+          delete styleTags[rawCss]
         }
       }
-    }, [css, namespace, styleElement, setStyleElement, isPostCss])
+    }, [css, namespace, isPostCss])
   }
 }
 
-function getStyleElement(oldStyleElement, isPostCss, css, namespace) {
+function getStyleElement(isPostCss, css, namespace, incrementKremingsIfFound = false) {
```

Comment:
Refactoring because we only want to increment the counter if we find the styleElement during initialization